### ISSUE TITLE
A11Y: MultiSelect component needs to indicate selected count.

### ISSUE THREAD

**Issue by sandhya-r-reddy** [carbon-design-system/carbon#2562]
Detailed description
JAWS screen reader didn't inform the user about the number of selected items , which displayed as part of clear selection/ selection count button. Screen reader user have no idea about the count of selected items when they select few items , then closes the list box and come back to the component .

And the tabbing instructions for screen reader is confusing, the listBox instructions are header before the combo box opening.

Is this issue related to a specific component?

MultiSelect

What did you expect to happen? What happened instead? What would you like to see changed?
Once the user selects multiple items and closes the list box, the select count should be indicated for screen reader user.
Ally Tabbing instructions for screen reader should be 1st for combo box and then list box.
What browser are you working in?
Firefox and Jaws Screen reader

What version of the Carbon Design System are you using?
6.39.3

What offering/product do you work on? Any pressing ship or release dates we should be aware of?
Genesis UI
